File: LeetCode/1001-1100/1014. Best Sightseeing Pair/src/Solution.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int prevMax = values[0];
        int max = Integer.MIN_VALUE;
        for (int j = 1; j < values.length; j++) {
            max = Math.max(max, prevMax + values[j] - j);
            prevMax = Math.max(prevMax, values[j] + j);
        }
        return max;
    }
}

File: LeetCode/1001-1100/1014. Best Sightseeing Pair/Solution.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& values) {
        int n = values.size();
        int maxScore = INT_MIN;
        int prevMax = values[0];
        for (int i = 1; i < n; ++i) {
            maxScore = max(maxScore, prevMax + values[i] - i);
            prevMax = max(prevMax, values[i] + i);
        }
        return maxScore;
    }
};